numpy/core/tests/test_ufunc.py

```diff
@@ -1964,3 +1964,28 @@ def test_ufunc_types(ufunc):
                 assert r.dtype == np.dtype(t)
         else:
             assert res.dtype == np.dtype(out)
+
+@pytest.mark.parametrize('ufunc', [getattr(np, x) for x in dir(np)
+                                if isinstance(getattr(np, x), np.ufunc)])
+def test_ufunc_noncontiguous(ufunc):
```

Comment:
a ufunc test for overlapping input and output memory is needed for vectorization too
`_gen_alignment_data` produces a bunch of data for unary and binary simd tests  (though not strided)
but it creates a huge amount of test cases which makes tests slow

nowadays atleast on x86 you don't need the alignment stuff anymore so maybe overlapping input and output for unary and binary functions is enough?

Reply:
Would it be enough to set `func(*args, out=arg[0])` or would you like to see offset overlap as well?